### PR TITLE
Fix list-objects returning nothing after restarting

### PR DESCRIPTION
**Pull request by camjackson**
Currently if fake-s3 is restarted, it always returns an empty response for the `list-objects` action. However, the objects are still there and can be fetched with `get-object` if you already know the key.

This PR is similar to [this one](https://github.com/jubos/fake-s3/pull/69), but including the extra fixes in comments from @Allactaga and @mark2997.

### CODE REVIEW

File: lib/fakes3/file_store.rb [hunk 1]
```diff
@@ -278,5 +278,11 @@ def create_metadata(content,request)
       end
       return metadata
     end
+     private
```
Comment (jubos) [inline lib/fakes3/file_store.rb]: Style nit: Indentation looks off here.

File: lib/fakes3/file_store.rb [hunk 2]
```diff
@@ -21,7 +21,8 @@ def initialize(root)
       @bucket_hash = {}
       Dir[File.join(root,"*")].each do |bucket|
         bucket_name = File.basename(bucket)
-        bucket_obj = Bucket.new(bucket_name,Time.now,get_objects(bucket_name, bucket))
+        puts "root bucket: #{bucket_name}"
```
Comment (jubos) [inline lib/fakes3/file_store.rb]: Leave the puts out to avoid console spew.

File: lib/fakes3/file_store.rb [hunk 3]
```diff
@@ -278,11 +279,24 @@ def create_metadata(content,request)
       end
       return metadata
     end
-     private
-     def get_objects bucket_name, path
-       Dir[File.join(path, '*')].map do |filepath|
-         get_object(bucket_name, File.basename(filepath), nil)
-       end
-     end 
+    private
```
Comment (jubos) [inline lib/fakes3/file_store.rb]: Add a newline here.

File: lib/fakes3/file_store.rb [hunk 4]
```diff
@@ -278,11 +279,24 @@ def create_metadata(content,request)
...
+    def get_objects bucket_name, path, root_path=nil
```
Comment (jubos) [inline lib/fakes3/file_store.rb]: Use 2 spaces for indentation and () in the method declaration like the other methods in the class.

**Comment by camjackson** [comment]
Thanks for your feedback @jubos! I actually made this PR from @saltzmanjoelh's fork (I didn't even know you could do a PR from a repo you don't own), which means I'm not able to push more code in order to amend this commit.

Instead I've opened a new PR, #221, which hopefully addresses your comments, and I'll close this one.

**Pull request closed without merging**